pkg/tests/ossm/smcp_install_test.go

```diff
@@ -15,219 +15,244 @@
 package ossm
 
 import (
-	"strings"
+	_ "embed"
+	"fmt"
 	"testing"
-	"time"
 
-	"github.com/maistra/maistra-test-tool/pkg/util"
+	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
 	"github.com/maistra/maistra-test-tool/pkg/util/env"
-	"github.com/maistra/maistra-test-tool/pkg/util/log"
-	"github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/hack"
+	"github.com/maistra/maistra-test-tool/pkg/util/oc"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
+	"github.com/maistra/maistra-test-tool/pkg/util/template"
+	. "github.com/maistra/maistra-test-tool/pkg/util/test"
 )
 
-func installDefaultSMCP23() {
-	log.Log.Info("Create SMCP v2.3 in ", meshNamespace)
-	util.ShellMuteOutputError(`oc new-project %s`, meshNamespace)
-	util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV23_template, smcp))
-	util.KubeApplyContents(meshNamespace, smmr)
-
-	// patch SMCP identity if it's on a ROSA cluster
-	if env.Getenv("ROSA", "false") == "true" {
-		util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
-	}
-	log.Log.Info("Waiting for mesh installation to complete")
-	util.Shell(`oc wait --for condition=Ready -n %s smcp/%s --timeout 300s`, meshNamespace, smcpName)
-	time.Sleep(time.Duration(20) * time.Second)
+type vars struct {
+	Name      string
+	Namespace string
 }
 
-func installDefaultSMCP22() {
-	log.Log.Info("Create SMCP v2.2 in ", meshNamespace)
-	util.ShellMuteOutputError(`oc new-project %s`, meshNamespace)
-	util.KubeApplyContents(meshNamespace, util.RunTemplate(smcpV22_template, smcp))
-	util.KubeApplyContents(meshNamespace, smmr)
-
-	// patch SMCP identity if it's on a ROSA cluster
-	if env.Getenv("ROSA", "false") == "true" {
-		util.Shell(`oc patch -n %s smcp/%s --type merge -p '{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}'`, meshNamespace, smcpName)
+func installDefaultSMCP(t TestHelper, ns string) {
+	version := env.Getenv("SMCPVERSION", "2.4")
+	t.Log("Tear down test case, install default SMCP version: %s", version)
+	// TODO: add the logic to install specific version from env variable SMCP version
+	oc.RecreateNamespace(t, ns)
+	vars := vars{
+		Name:      smcp.Name,
+		Namespace: smcp.Namespace,
+	}
+	oc.ApplyString(t, meshNamespace, template.Run(t, smcpV24_template, vars))
+	oc.ApplyString(t, meshNamespace, smmr)
+	if env.IsRosa() {
+		oc.PatchWithMerge(
+			t, meshNamespace,
+			fmt.Sprintf("smcp/%s", smcpName),
+			`{"spec":{"security":{"identity":{"type":"ThirdParty"}}}}`)
 	}
-	log.Log.Info("Waiting for mesh installation to complete")
-	util.Shell(`oc wait --for condition=Ready -n %s smcp/%s --timeout 300s`, meshNamespace, smcpName)
-	time.Sleep(time.Duration(20) * time.Second)
+	oc.WaitSMCPReady(t, meshNamespace, smcpName)
+	oc.WaitCondition(t, meshNamespace, "smcp", smcpName, "Ready")
+	t.LogStep("Check SMCP is Ready")
 }
 
```

Comment:
Why do we need this function? If it's just to make the test cleaner, then why don't we have similar functions for all the other SMCP versions?

Reply:
Is used in the clean-up and is just to reset the default smcp, if we have a failure making an upgrade this func will to cleanup and reinstall the default version (In our case is 2.4 because the branch, is not prepared to run against all the smcp version )

Reply:
Ah, sorry, I didn't notice that it was only used in the cleanup. I thought it was used to install24 also...

But I don't think this function should be here, since we already have a function that performs this setup. And that function will probably change. And whoever changes it, won't know that they need to also change this function here. That's why this function here shouldn't exist. Instead, the cleanup code should call the other function.